src/compression.rs

```diff
@@ -0,0 +1,226 @@
+//! Compression-related methods, all used in `axoasset::Local`
+
+use crate::error::*;
+use camino::Utf8Path;
+use flate2::{write::ZlibEncoder, Compression, GzBuilder};
+use std::{
+    fs::{self, DirEntry},
+    io::BufReader,
+};
+use xz2::write::XzEncoder;
+use zip::ZipWriter;
+
+/// Internal tar-file compression algorithms
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub(crate) enum CompressionImpl {
```

Comment:
I figured it'd be best to expose all possible methods, while at the same time keeping a crate-internal duplicate of `cargo-dist`s `CompressionImpl` so we can carry over the function code mostly unmodified (with the exception of the error handling)